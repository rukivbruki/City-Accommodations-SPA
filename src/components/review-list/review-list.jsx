import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import Review from "../review/review";
import {Operation} from "../../reducer/data/data";
import {getReviews} from "../../reducer/data/selector";
import withLoadData from "../../hocs/with-load-data/with-load-data";

class ReviewList extends PureComponent {
  render() {
    const {reviews} = this.props;

    return (
      <>
        <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
        <ul className="reviews__list">
          {reviews.map((review) => {
            return <Review {...review} key={review.id}/>;
          })}
        </ul>
      </>
    );
  }
}

ReviewList.propTypes = {
  reviews: PropTypes.array.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  const reviews = getReviews(state, ownProps.id);
  return {
    reviews,
    isRequiredDataLoad: Boolean(!reviews),
  };
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  onLoadData: () => dispatch(Operation.loadReviews(ownProps.id))
});

const ReviewListWrapped = withLoadData(ReviewList);
export {ReviewList};
export default connect(mapStateToProps, mapDispatchToProps)(ReviewListWrapped);
