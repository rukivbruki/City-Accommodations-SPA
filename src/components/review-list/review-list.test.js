import React from "react";
import {configure, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import {ReviewList} from "./review-list";

configure({adapter: new Adapter()});

it(`list of reviews renders correctly`, () => {
  const review = shallow(<ReviewList
    isRequiredDataLoad={false}
    onLoadData={jest.fn()}
    reviews={[]}
    id={1}
  />);

  expect(review).toMatchSnapshot();
});
