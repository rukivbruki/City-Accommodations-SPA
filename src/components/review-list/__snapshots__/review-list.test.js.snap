// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list of reviews renders correctly 1`] = `
<Fragment>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      0
    </span>
  </h2>
  <ul
    className="reviews__list"
  />
</Fragment>
`;
